### PR TITLE
k8s 1.24 ci updates [1.13]

### PR DESCRIPTION
**Pull request by jenshu**
more k8s 1.24 updates:
- ci
- docs
- kubectl image

### CODE REVIEW

File: jobs/kubectl/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM bitnami/kubectl:1.22.12 as kubectl
+FROM bitnami/kubectl:1.24.7 as kubectl
```
Comment (sam-heilbron) [inline jobs/kubectl/Dockerfile:1]: Do we want this version to match our client libs or are server version? I had imagined we would keep this in line with our client libraries

**Review by nfuden — APPROVED** [review]

File: jobs/kubectl/Dockerfile (hunk 1 above)
Comment (nfuden) [reply jobs/kubectl/Dockerfile:1]: I am unsure on this but we arent bumping k8s due to go dependency bump but at the same time if being deployed in ci as 1.24 env it seems to make sense to be deployed as that?

Comment (jenshu) [reply jobs/kubectl/Dockerfile:1]: spoke offline; we decided to keep this change so it matches the server version we are testing against

**Review by elcasteel — APPROVED** [review]

**Pull request merged**